pkg/backends/zenoss/zenoss_test.go

```diff
@@ -1,7 +1,311 @@
 package zenoss
 
-import "testing"
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
 
-func TestNothing(t *testing.T) {
-	t.Error("no tests")
+	"google.golang.org/grpc"
+
+	"github.com/atlassian/gostatsd"
+	_struct "github.com/golang/protobuf/ptypes/struct"
+	"github.com/stretchr/testify/assert"
+	proto "github.com/zenoss/zenoss-protobufs/go/cloud/data_receiver"
+)
+
+// mocking connection object, to process requests locally
+type connectionMock struct {
+	putMetricsErrors int
+	putModelsErrors  int
+
+	putMetricsRequests int
+	putModelsRequests  int
+
+	metricsBody *proto.Metrics
+	modelsBody  *proto.Models
+}
+
+// make request and check it for errors
+func makeRequest(t *testing.T, c Client, cb func() *gostatsd.MetricMap) {
+	res := make(chan []error, 1)
+	c.SendMetricsAsync(context.Background(), cb(), func(errs []error) {
+		res <- errs
+	})
+	errs := <-res
+	for _, err := range errs {
+		assert.NoError(t, err)
+	}
+}
+
+func TestMetricsRetries(t *testing.T) {
+	requestsWithError := 3
+	client := newTestClient(requestsWithError, defaultMetricsPerBatch)
+	makeRequest(t, client, func() *gostatsd.MetricMap { return twoCounters() })
+
+	expectedRequests := 4
+	assert.Equal(t, expectedRequests, client.client.(*connectionMock).putMetricsRequests)
```

Comment:
might be more readable if you provided a nicer syntax for client.client.(*connectionMock) in these assertions.  Maybe just set a local variable and use that?